syntax = "proto3";

option go_package = "jd.com/jd-infer/xllm;xllm";
package xllm.proto;
option cc_enable_arenas = true;
option cc_generic_services = true;

import "completion.proto";
import "chat.proto";
import "embedding.proto";
import "image_generation.proto";
import "rerank.proto";
import "models.proto";
import "anthropic.proto";

message HttpRequest {
};

message HttpResponse {
};

service XllmAPIService {
  rpc Completions (CompletionRequest) returns (CompletionResponse);
  rpc CompletionsHttp (HttpRequest) returns (HttpResponse);

  rpc ChatCompletions (ChatRequest) returns (ChatResponse);
  rpc ChatCompletionsHttp (HttpRequest) returns (HttpResponse);

  rpc Embeddings (EmbeddingRequest) returns (EmbeddingResponse);
  rpc EmbeddingsHttp (HttpRequest) returns (HttpResponse);

  rpc ImageGeneration(ImageGenerationRequest) returns (ImageGenerationResponse);
  rpc ImageGenerationHttp(HttpRequest) returns (HttpResponse);

  rpc Rerank (RerankRequest) returns (RerankResponse);
  rpc RerankHttp (HttpRequest) returns (HttpResponse);
  
  rpc Models (ModelListRequest) returns (ModelListResponse);
  rpc ModelsHttp (HttpRequest) returns (HttpResponse);

  rpc ModelVersionsHttp (HttpRequest) returns (HttpResponse);

  rpc AnthropicMessagesHttp (HttpRequest) returns (HttpResponse);
};
